apps/miranum-bpmn-modeler-webview/src/main.ts

```diff
@@ -340,4 +323,63 @@ function asyncDebounce<F extends (...args: any[]) => Promise<any>>(
         }) as ReturnType<F>;
 }
 
+function createBpmnModeler(executionPlatformVersion: ExecutionPlatformVersion): BpmnModeler {
+    let bpmnModeler;
+    switch (executionPlatformVersion) {
+        case ExecutionPlatformVersion.None:
+        case ExecutionPlatformVersion.Camunda7: {
+            bpmnModeler = new BpmnModeler({
+                container: "#js-canvas",
+                keyboard: {
+                    bindTo: document,
+                },
+                propertiesPanel: {
+                    parent: "#js-properties-panel",
+                },
+                additionalModules: [
+                    // Properties Panel
+                    BpmnPropertiesPanelModule,
+                    BpmnPropertiesProviderModule,
+                    CamundaPlatformPropertiesProviderModule,
+                    CamundaPlatformBehaviors,
+                    miragonProviderModule,
+                    // Element Templates
+                    ElementTemplatesPropertiesProviderModule,
+                    ElementTemplateChooserModule,
+                    // Other Plugins
+                    TokenSimulationModule,
+                ],
+                moddleExtensions: {
+                    camunda: camundaModdleDescriptors,
+                },
+            });
+            break;
+        }
+        case ExecutionPlatformVersion.Camunda8: {
+            bpmnModeler = new BpmnModeler({
+                container: "#js-canvas",
+                keyboard: {
+                    bindTo: document,
+                },
+                propertiesPanel: {
+                    parent: "#js-properties-panel",
+                },
+                additionalModules: [
+                    // Properties Panel
+                    BpmnPropertiesPanelModule,
+                    BpmnPropertiesProviderModule,
+                    ZeebePropertiesProviderModule,
+                    // Other Plugins
+                    TokenSimulationModule,
+                ],
+                moddleExtensions: {
+                    camunda: ZeebeBpmnModdle,
+                },
+            });
+            break;
+        }
+    }
+    return bpmnModeler;
```

Comment:
I maybe would rather put it in a differently named function. CreateBPMNModeler serves one reason (creating the modeler object) and would not mix it with adding event listeners

Reply:
👍🏻 Also fine by me. But we have to call this method before we call *openXML()* because then we never get an error when we open the modeler the first time.